pkg/cmd/component/init.go

```diff
@@ -0,0 +1,317 @@
+package bootstrapnew
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/jenkins-x/go-scm/scm"
+	"github.com/jenkins-x/go-scm/scm/factory"
+	"github.com/openshift/odo/pkg/log"
+	gitops "github.com/redhat-developer/gitops-generator/pkg"
+	"github.com/redhat-developer/kam/pkg/cmd/component/cmd/ui"
+	"github.com/redhat-developer/kam/pkg/cmd/genericclioptions"
+	"github.com/redhat-developer/kam/pkg/cmd/utility"
+	"github.com/redhat-developer/kam/pkg/pipelines/accesstoken"
+	pipelines "github.com/redhat-developer/kam/pkg/pipelines/component"
+	"github.com/redhat-developer/kam/pkg/pipelines/ioutils"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/zalando/go-keyring"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	InitRecommendedCommandName = "init"
+	defaultRepoDescription     = "Bootstrapped GitOps Repository based on Components"
+	applicationFolderFlag      = "application-folder"
+)
+
+var (
+	initExample = ktemplates.Examples(`
+   # Init command to initialize GitOps repository.
+       kam init --application-folder <path to application> --git-repo-url <https://github.com/<your organization>/gitops.git --secret <your git access token>
+      
+   %[1]s
+   `)
+
+	initLongDesc  = ktemplates.LongDesc(`Git intialize the GitOps repository.`)
+	initShortDesc = `Preform the Git init, branch and remote commands to initialize the GitOps folder.`
+)
+
+func NewInitParameters() *InitParameters {
+	return &InitParameters{
+		GeneratorOptions: &pipelines.GeneratorOptions{},
+	}
+}
+
+type InitParameters struct {
+	*pipelines.GeneratorOptions
+	Interactive bool
+}
+
+// checks the application folder constraints for non interactive mode
+// if the application folder path is present then it check wether it is valid for application
+// if the folder path is not present check for valid application name to create it
+
+func CheckApplicationPath(app afero.Afero, path string) error {
```

Comment:
In the ui.go changes, we have a function that does something similar.  Could we combine/simplify this?

Reply:
Yes the function is similar for sure. But in the ui.go it is for the prompt so we need to have for non-interactive mode.
As after validating the next prompt we add for is confirmation.